internal/pkg/mining/worker.go

```diff
@@ -247,16 +236,108 @@ func (w *DefaultWorker) Mine(ctx context.Context, base block.TipSet, nullBlkCoun
 		outCh <- Output{Err: err}
 		return
 	}
-	weHaveAWinner, err := w.election.DeprecatedIsElectionWinner(ctx, powerTable, electionTicket, nullBlkCount, electionProof, workerAddr, w.minerAddr)
+	sortedSectorInfos, err := powerTable.SortedSectorInfos(ctx, w.minerAddr)
```

Comment:
this entire method could use a bit more grouping of sections of code and comments

Reply:
10,000% yes
I'm hoping to clean this up in the near future.

Reply:
The error handling is absolute trash too